Break most of Perseus's dependencies on KaTeX

benchristel opened the pull request:
We are replacing KaTeX with MathJax 3 in webapp and the mobile app. Before we
can do that, Perseus needs to not depend on KaTeX at all.

This PR doesn't fully remove references to KaTeX from Perseus. It is just the
first step.

Issue: https://khanacademy.atlassian.net/browse/LC-459

## Test plan:

Inspect all Storybook stories and verify that e.g. the Graphie graphs with
rendered math expressions look good.

benchristel commented:
@jeremywiebe it's going to be easiest to review this one commit at a time, I think.

benchristel commented on packages/perseus/src/components/sortable.tsx:
```diff
@@ -432,15 +427,13 @@ class Sortable extends React.Component<SortableProps, SortableState> {
         margin: 5,
         onChange: function () {},
         linterContext: PerseusLinter.linterContextDefault,
-        waitForKatexLoad: true,
```
@jeremywiebe we removed `waitForKatexLoad` during our pairing session, but I think removing it changes behavior, since it defaulted to `true`. Do we care about keeping this functionality? I have an idea for how to make it work:

- Render a hidden`TeX` component in `Sortable`
- Provide an `onRender` callback to the `TeX` that sets a `texReady` state variable on `Sortable`
- Render a spinner until `texReady` is true.

benchristel replied:
I don't think we can guarantee that MathJax is loaded before the sortable renders. At least in the first incremental rollout I've planned for MathJax 3 (using it as the fallback instead of MathJax 2), MathJax will be behind a loadable, since it won't be needed on most pages.

I think we ought to keep this functionality and just change the name and implementation.

jeremywiebe reviewed (approved):
Sorry it took so long to get to this.  I think it looks good though!

benchristel commented:
@jeremywiebe I added back the `waitForKaTeXLoad` prop to Sortable and renamed it to `waitForTexRendererToLoad`. Do you want to take another look?

Also, should this PR have a changeset? When do we want to release this?

Note that this is a breaking change, in the sense that clients will have to remove the no-longer-supported properties from their Dependencies when they upgrade Perseus to this version.

jeremywiebe commented on packages/perseus/src/components/sortable.tsx:
```diff
@@ -584,15 +576,34 @@ class Sortable extends React.Component<SortableProps, SortableState> {
+        // has successfully rendered a test element. If we didn't do this, the user
+        // might see a sortable with empty cells on first render, and then the math
+        // would pop in a few moments later once the rendering library loaded.
+        if (
+            this.props.waitForTexRendererToLoad &&
+            !this.state.texRendererLoaded
+        ) {
+            const {TeX} = getDependencies();
+            return (
+                <>
+                    <CircularSpinner />
+                    <div style={{display: "none"}}>
+                        <TeX
+                            onRender={() =>
+                                this.setState({texRendererLoaded: true})
+                            }
+                        >
+                            1
+                        </TeX>
+                    </div>
+                </>
+            );
```
I'm confused by this.  We render the spinner _and_ the `<TeX>`.  Shouldn't we be switching on the `waitForTexRendererToLoad` and `texRendererLoaded` flags and rendering a spinner else the `TeX`?

benchristel replied:
We're using this TeX component to tell the TeX renderer to load and detect when it has loaded. Once it renders, we set `texRendererLoaded` to `true`.

jeremywiebe reviewed (commented):
> Also, should this PR have a changeset? When do we want to release this?

Yes, please add a changeset.  Use `yarn changeset` and follow the prompts. 

> Note that this is a breaking change, in the sense that clients will have to remove the no-longer-supported properties from their Dependencies when they upgrade Perseus to this version.

That's fine. That means this'll be a major version bump for Perseus then. 

I'm holding off on approving for the sorter widget question I have. I'm probably just not following something obvious but want to double-check. :)

jeremywiebe reviewed (approved):
Thanks for explaining on a call! :)
